barretenberg/cpp/src/barretenberg/stdlib/primitives/group/straus_scalar_slice.cpp

```diff
@@ -30,82 +31,56 @@ straus_scalar_slice<Builder>::straus_scalar_slice(Builder* context,
                                                   const size_t table_bits)
     : _table_bits(table_bits)
 {
-    using FF = typename Builder::FF;
-    using witness_t = stdlib::witness_t<Builder>;
-
-    constexpr bool IS_ULTRA = Builder::CIRCUIT_TYPE == CircuitType::ULTRA;
-
     // convert an input cycle_scalar object into a vector of slices, each containing `table_bits` bits.
     // this also performs an implicit range check on the input slices
-    const auto slice_scalar = [&](const field_t& scalar, const size_t num_bits) {
+    const auto compute_scalar_slices =
+        [&](const field_t& scalar, const size_t num_bits) -> std::pair<std::vector<field_t>, std::vector<uint64_t>> {
+        const size_t num_slices = numeric::ceil_div(num_bits, table_bits);
+
         // we record the scalar slices both as field_t circuit elements and u64 values
         // (u64 values are used to index arrays and we don't want to repeatedly cast a stdlib value to a numeric
         // primitive as this gets expensive when repeated enough times)
-        std::pair<std::vector<field_t>, std::vector<uint64_t>> result;
```

Comment:
Here I'm just trying to make this code a bit more explicit by splitting this `pair` into two separate vectors with meaningful names, then constructing the original `pair` on return.